mt-st

```diff
@@ -0,0 +1,39 @@
+#mt bash completion by Paweł Marciniak <sunwire<at>gmail.com>
+#
+
+_mt () {
+    local cur prev words cword
+    _init_completion || return
+
+    #possible commands
+    commands="weof wset eof fsf fsfm bsf bsfm fsr bsr fss bss rewind offline rewoffl eject retension eod seod seek tell status erase setblk lock unlock load compression setdensity drvbuffer stwrthreshold stoptions stsetoptions stclearoptions defblksize defdensity defdrvbuffer defcompression stsetcln sttimeout stlongtimeout densities setpartition mkpartition partseek asf stshowopt"
+
+    COMPREPLY=()
+
+    case $prev in
+        -v | --version)
+            return
+            ;;
+        -f )
+            if [ -x "$(command -v udevadm)" ]; then
+                #list tape devices
+                for d in `udevadm trigger --verbose --dry-run --type=devices --subsystem-match=scsi_tape`; do
```

Comment:
Output of this on my system is:

```
/sys/devices/pci0000:00/0000:00:03.2/0000:0a:00.0/host0/port-0:4/end_device-0:4/target0:0:4/0:0:4:0/scsi_tape/nst0
/sys/devices/pci0000:00/0000:00:03.2/0000:0a:00.0/host0/port-0:4/end_device-0:4/target0:0:4/0:0:4:0/scsi_tape/nst0a
/sys/devices/pci0000:00/0000:00:03.2/0000:0a:00.0/host0/port-0:4/end_device-0:4/target0:0:4/0:0:4:0/scsi_tape/nst0l
/sys/devices/pci0000:00/0000:00:03.2/0000:0a:00.0/host0/port-0:4/end_device-0:4/target0:0:4/0:0:4:0/scsi_tape/nst0m
/sys/devices/pci0000:00/0000:00:03.2/0000:0a:00.0/host0/port-0:4/end_device-0:4/target0:0:4/0:0:4:0/scsi_tape/st0
/sys/devices/pci0000:00/0000:00:03.2/0000:0a:00.0/host0/port-0:4/end_device-0:4/target0:0:4/0:0:4:0/scsi_tape/st0a
/sys/devices/pci0000:00/0000:00:03.2/0000:0a:00.0/host0/port-0:4/end_device-0:4/target0:0:4/0:0:4:0/scsi_tape/st0l
/sys/devices/pci0000:00/0000:00:03.2/0000:0a:00.0/host0/port-0:4/end_device-0:4/target0:0:4/0:0:4:0/scsi_tape/st0m
```

And it seems you're just taking the last part of the path. I think this can be easier retrieved by going purely via sysfs, i.e.:

```
for tape in /sys/class/scsi_tape/*; do devs+="/dev/${tape##*/} "; done;
```

Without needing to either check for udevadm, or run the command. What do you think?

Reply:
It looks good to me. We can use sysfs instead of udevadm.